### ISSUE TITLE
np.concatenate((1 - y, y), -1)?

### ISSUE THREAD

**Issue by ZaherHsu** [imlab-uiip/lung-segmentation-3d#10]
hi. eclique. After reading your code, I don't understand much about it. I hope to get your advice. I don't quite understand 'np.concatenate (1-y, y),-1)',What does this code mean?and Why do you want to do the reverse operation?I implore your answer.
    y = np.expand_dims(y, -1)
    y = np.array(y)
    y = np.concatenate((1 - y, y), -1)
    y = np.array(y)

**Comment by eclique** [comment]
This is done since the model outputs two channels: _no lungs_ and _lungs_. 
This code takes the lung mask, say 1 - lungs, 0 - no lungs, and concatenates it along last axis. Then loss can be computed using cross-entropy between model's output and ground truth constructed in such way (1-y, y). 

In retrospective, it would probably make more sense to output one output channel and use sigmoid instead of softmax.

**Comment by ZaherHsu** [comment]
Thank you for your answer, you helped me solve a lot of problems, but now I have a new problem, I found that when the addition of. Nii files more than 1, 024, the program will have problems with [Error 24] Too many open files. Have you ever had such a question at that time, is it convenient to share it with me?

**Comment by eclique** [comment]
@LeoXuZH no, I haven't had this error. You need to close the open files somewhere, depending on the way you open them.

**Comment by fredsabuni** [comment]
Hello
@eclique I am having an error the code just crash out of no where. I updated the code and start using Tensorflow v2 with keras

**Comment by eclique** [comment]
Hi. Unfortunately, I can't help with the code adaptation for newer versions of tf and keras.
